airbyte-db/lib/src/main/java/io/airbyte/db/init/DatabaseInitializer.java

```diff
@@ -0,0 +1,144 @@
+      if (dslContext.isPresent()) {
+        final Database database = new Database(dslContext.get());
+        new ExceptionWrappingDatabase(database).transaction(ctx -> {
+          // Verify that all the required tables are present
+          if (getTableNames().stream().allMatch(tableName -> hasTable(dslContext.get(), tableName))) {
+            getLogger().info("The {} database is initialized", getDatabaseName());
+          } else {
+            getLogger().info("The {} database has not been initialized; initializing it with schema: \n{}", getDatabaseName(), getInitialSchema());
+            ctx.execute(getInitialSchema());
+            getLogger().info("The {} database successfully initialized with schema: \n{}.", getDatabaseName(), getInitialSchema());
+          }
+
+          // Always return true, as the tables either exist or were just initialized by this transaction
+          return true;
+        });
+      } else {
+        throw new InterruptedException("Database configuration not present.");
+      }
+    } else {
+      throw new InterruptedException("Availability check not configured.");
+    }
+  }
+
+  /**
+   * Tests whether the provided table exists in the database.
+   *
+   * @param ctx A {@link DSLContext} used to query the database.
+   * @param tableName The name of the table.
+   * @return {@code True} if the table exists or {@code false} otherwise.
+   */
+  default boolean hasTable(final DSLContext ctx, final String tableName) {
+    return ctx.fetchExists(select()
+        .from("information_schema.tables")
+        .where(DSL.field("table_name").eq(tableName)
+            .and(DSL.field("table_schema").eq("public"))));
+  }
+
+  /**
+   * Retrieves the {@link DatabaseAvailabilityCheck} used to verify that the database is running and
+   * available.
+   *
+   * @return The {@link DatabaseAvailabilityCheck}.
+   */
+  Optional<DatabaseAvailabilityCheck> getDatabaseAvailabilityCheck();
+
+  /**
+   * Retrieves the configured database name to be tested.
+   *
+   * @return The name of the database to test.
+   */
+  String getDatabaseName();
+
+  /**
+   * Retrieves the configured {@link DSLContext} to be used to test the database availability.
+   *
+   * @return The configured {@link DSLContext} object.
+   */
+  Optional<DSLContext> getDslContext();
+
+  /**
+   * Retrieve the initial schema to be applied to the database if the database is not already
+   * populated with the expected table(s).
+   *
+   * @return The initial schema.
+   */
+  String getInitialSchema();
+
+  /**
+   * Retrieves the configured {@link Logger} object to be used to record progress of the migration
+   * check.
+   *
+   * @return The configured {@link Logger} object.
+   */
+  Logger getLogger();
+
+  /**
+   * The collection of table names that will be used to confirm database availability.
+   *
+   * @return The collection of database table names.
+   */
+  default Collection<String> getTableNames() {
+    return List.of();
+  }
```

Comment:
Is it necessary to have this default implementation? It seems like this could lead to issues if someone implements this class without overriding this method, causing the initialize logic to basically be a no-op

Reply:
This is more temporary code to avoid the possibility that the implementation doesn't set the list or override it.